### PR TITLE
nrf_modem: bindtopdn

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Add bindtopdn socket option.

### CODE REVIEW

File: nrf_modem/doc/sockets.rst [hunk 1]
```diff
@@ -52,7 +52,7 @@ The following table shows all socket options supported by the Modem library.
 +-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
 | NRF_SOL_SOCKET  | NRF_SO_SNDTIMEO                 | ``struct nrf_timeval`` | get/set    | Timeout value for a socket send operation.                                                 |
 +-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
-| NRF_SOL_SOCKET  | NRF_SO_BINDTODEVICE             | ``char *``             | set        | Bind this socket to a specific PDN like ``pdn0`` as specified in the passed option value.  |
+| NRF_SOL_SOCKET  | NRF_SO_BINDTOPDN                | ``char *``             | set        | Bind this socket to a specific PDN ID.                                                     |
```
Comment (lemrey) [inline nrf_modem/doc/sockets.rst:55]: This takes an `int` now :)

Comment (eivindj-nordic) [reply nrf_modem/doc/sockets.rst]: Thanks, updated :)

File: nrf_modem/doc/sockets.rst [hunk 2]
```diff
@@ -340,15 +340,16 @@ For more information about how to configure PDP contexts, activate PDN connectio
 Configuring a socket to use a PDN
 =================================
 
-The application can select which PDN to use on a specific socket by using the :c:func:`nrf_setsockopt` function, with the :c:macro:`NRF_SO_BINDTODEVICE` option and specifying the PDN ID as a string, prefixed by ``pdn``.
-For example, to select the PDN with ID 1, the application must pass ``pdn1`` as the option value.
+The application can select which PDN to use on a specific socket by using the :c:func:`nrf_setsockopt` function, with the :c:macro:`NRF_SO_BINDTOPDN` option and specifying the PDN ID as an integer.
```
Comment (MirkoCovizzi) [inline nrf_modem/doc/sockets.rst:343]:
```suggestion
The application can select which PDN to use on a specific socket by using the :c:func:`nrf_setsockopt` function, with the :c:macro:`NRF_SO_BINDTOPDN` socket option and specifying the PDN ID as an integer.
```

File: nrf_modem/doc/sockets.rst [hunk 3]
```diff
@@ -52,7 +52,7 @@ The following table shows all socket options supported by the Modem library.
 +-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
 | NRF_SOL_SOCKET  | NRF_SO_SNDTIMEO                 | ``struct nrf_timeval`` | get/set    | Timeout value for a socket send operation.                                                 |
 +-----------------+---------------------------------+------------------------+------------+--------------------------------------------------------------------------------------------+
-| NRF_SOL_SOCKET  | NRF_SO_BINDTODEVICE             | ``char *``             | set        | Bind this socket to a specific PDN like ``pdn0`` as specified in the passed option value.  |
+| NRF_SOL_SOCKET  | NRF_SO_BINDTOPDN                | ``int``             | set        | Bind this socket to a specific PDN ID.                                                     |
```
Comment (lemrey) [inline nrf_modem/doc/sockets.rst:55]: Check the table here.. do we need to add some spaces?

**Review by lemrey — APPROVED** [review]
Looks good otherwise

**Pull request merged**